### ISSUE TITLE
Detach the hidden states

### ISSUE THREAD

**Issue by paulomann** [dpressel/dlss-tutorial#1]
Dear Daniel Pressel,

I have found your tutorial for the International Summer School on Deep Learning very elucidating (I was not there, though). However, there is one concept that is not clear to me. For example, when you say in the Second Section (Contextualized Embeddings) in the colab:

> At every step of training, we will detach our hidden states, preventing full backpropagation, but we will initialize the new batch from our old hidden state

Could you further elaborate on why detaching the hidden states (both h, and c) is a necessary step? Or could you point to a reference where this is explained?

Thanks!

**Comment by dpressel** [comment]
Hi, sorry for the late response.  We are doing [Truncated Backpropagation Through Time](https://www.cs.utoronto.ca/~ilya/pubs/ilya_sutskever_phd_thesis.pdf) -- basically, we are going to limit how far back in time we will backpropagate for the LSTM, but we will allow the hidden state to be initialized in each truncated sample from the previous hidden state.  See this SO post:

https://stackoverflow.com/questions/53912956/truncated-backpropagation-through-time-bptt-in-pytorch
